### ISSUE TITLE
workspace-search UI Improvements

### ISSUE THREAD

**Issue by mikeharv** [RaspberryPiFoundation/blockly#10402]
### Check for duplicates

- [ ] I have searched for similar issues before opening a new one.

### Problem

Follow-up to https://github.com/RaspberryPiFoundation/blockly/issues/10324

The current UI is hard to parse. Blocks matching the search query are black and the current match is grey. This screen shot shows highlighting the second match for "if":

<img width="339" height="357" alt="Image" src="https://github.com/user-attachments/assets/d72aac1d-409a-4018-83ed-84ce366bb61d" />

### Request

Maribeth suggested having the search input element take ephemeral focus so that the matched blocks could have focus outlines, making them more obvious to sighted users. See: https://github.com/RaspberryPiFoundation/blockly/pull/10316#discussion_r3881936456

Additionally, it is worth exploring other improvements to the UI. Inspiration could be taken from other search UI that indicates multiple visual matches simultaneously.

<img width="889" height="702" alt="Image" src="https://github.com/user-attachments/assets/17065b59-128a-4a13-9654-3fa3a11bd0d8" />
